### ISSUE TITLE
Handle NA values during HPO 

### ISSUE THREAD

**Issue by borauyar** [BIMSBbioinfo/flexynesis#84]
During hyperparameter optimisation, some parameter setup may lead to exploding/vanishing gradients that yield NA loss values. This kills the training, but it should be allowed to continue the next hpo iteration with a different parameter combination. In such cases, setting the loss value to a high number should solve the issue, so the optimization avoids that parameter combination and training continues.
